pkg/fleet/internal/cdn/cdn.go

```diff
@@ -50,87 +46,105 @@ type orderConfig struct {
 }
 
 // New creates a new CDN.
-func New(env *env.Env) *CDN {
-	return &CDN{
-		env: env,
+func New(env *env.Env) (*CDN, error) {
+	config := pkgconfigsetup.Datadog()
+	detectenv.DetectFeatures(config)
+	client, err := remoteconfig.NewHTTPClient(
+		config,
+		fmt.Sprintf("https://remote-config.%s", env.Site),
+		fmt.Sprintf("https://config.%s", env.Site),
+		env.Site,
+		env.APIKey,
+		"",
+		version.AgentVersion,
+	)
+	if err != nil {
+		return nil, err
 	}
+
+	return &CDN{
+		client:                client,
+		currentTargetsVersion: 1,
+		currentRootsVersion:   1,
+	}, nil
 }
 
 // Get gets the configuration from the CDN.
 func (c *CDN) Get(ctx context.Context) (_ *Config, err error) {
-	span, ctx := tracer.StartSpanFromContext(ctx, "cdn.Get")
+	span, _ := tracer.StartSpanFromContext(ctx, "cdn.Get")
 	defer func() { span.Finish(tracer.WithError(err)) }()
-	configLayers, err := c.getOrderedLayers(ctx)
+	configLayers, err := c.getOrderedLayers()
 	if err != nil {
 		return nil, err
 	}
 	return newConfig(configLayers...)
 }
 
 // getOrderedLayers calls the Remote Config service to get the ordered layers.
-// Today it doesn't use the CDN, but it should in the future
-func (c *CDN) getOrderedLayers(ctx context.Context) ([]*layer, error) {
-	// HACK(baptiste): Create a dedicated one-shot RC service just for the configuration
-	// We should use the CDN instead
-	config := pkgconfigsetup.Datadog()
-	detectenv.DetectFeatures(config)
-	hostname, err := pkghostname.Get(ctx)
+func (c *CDN) getOrderedLayers() ([]*layer, error) {
+	agentConfigUpdate, err := c.client.GetCDNConfigUpdate(
+		[]string{"AGENT_CONFIG"},
+		// Always send 0 since we are relying on the CDN cache state instead of our own tracer cache. This will fetch the latest configs from the cache/CDN everytime.
+		0,
+		// Not using the roots; send the highest seen version of roots so don't received them all on every request
+		c.currentRootsVersion,
+		// Not using a client cache; fetch all the applicable target files every time.
+		[]*pbgo.TargetFileMeta{},
```

Comment:
Would it be possible to remove/use defaults for these the three arguments if there is always some assumed behavior

Reply:
Sadly Go doesn't support optional parameters, which would be ideal for something like this. In this case, I do want to keep these parameters though, as there can (and likely will be) clients who will want to do "caching" of the current config state, for which these params would be used.